File: MovieProject/MovieProject.Web/Modules/Northwind/OrderDetailsExtended/OrderDetailsExtendedForm.cs
using Serenity.ComponentModel;

namespace MovieProject.Northwind.Forms;

[FormScript("Northwind.OrderDetailsExtended")]
[BasedOnRow(typeof(OrderDetailsExtendedRow), CheckNames = true)]
public class OrderDetailsExtendedForm
{
    public int ProductId { get; set; }
    public string ProductName { get; set; }
    public decimal UnitPrice { get; set; }
    public short Quantity { get; set; }
    public float Discount { get; set; }
    public decimal ExtendedPrice { get; set; }
}

File: MovieProject/MovieProject.Web/Modules/ServerTypes/Northwind/OrderDetailsExtendedForm.ts
import { IntegerEditor, StringEditor, DecimalEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface OrderDetailsExtendedForm {
    ProductId: IntegerEditor;
    ProductName: StringEditor;
    UnitPrice: DecimalEditor;
    Quantity: IntegerEditor;
    Discount: DecimalEditor;
    ExtendedPrice: DecimalEditor;
}

export class OrderDetailsExtendedForm extends PrefixedContext {
    static readonly formKey = 'Northwind.OrderDetailsExtended';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!OrderDetailsExtendedForm.init)  {
            OrderDetailsExtendedForm.init = true;

            var w0 = IntegerEditor;
            var w1 = StringEditor;
            var w2 = DecimalEditor;

            initFormType(OrderDetailsExtendedForm, [
                'ProductId', w0,
                'ProductName', w1,
                'UnitPrice', w2,
                'Quantity', w0,
                'Discount', w2,
                'ExtendedPrice', w2
            ]);
        }
    }
}